src/utils/SortingNetworkGenerator.java

```diff
@@ -99,6 +107,14 @@ public static String encodeNetworkAndDisplay(String name, Integer[] indices, int
             indicesInt[i] = indices[i];
         }
         if (!encodeNetwork(indicesInt, path)) {
+			if (indicesInt.length == 0) {
+				JOptionPane.showMessageDialog(null, "Sort does not compare indices; An empty sorting network cannot be generated.",
+					"File not saved", JOptionPane.ERROR_MESSAGE);
+			}
+			else {
+				JOptionPane.showMessageDialog(null, "Sorting network too large. (Exceeds " + formatter.format(LIMIT) + " comparators)",
+					"File not saved", JOptionPane.ERROR_MESSAGE);
```

Comment:
One thing of note is that `indices.length` is actually twice the number of comparators, so the limit here is actually fairly small and will make things like bose-nelson not generate. I think a warning over a hard limit would suffice.

Reply:
if the limit is too low then u can always increase it i dont see why some1 would want the program to just freeze when the network is too large

Reply:
Because people don't necessarily want to edit the code (especially if they have the JAR and have to technical experience). Either way, a limit of 3500 comparisons is too small.

Reply:
i originally intended the limit to be 7000 btw i forgot these were pairs